api/src/main/java/org/apache/cloudstack/api/command/user/template/CreateTemplateCmd.java

```diff
@@ -251,26 +267,33 @@ public long getEntityOwnerId() {
             }
         }
 
-        if(projectId != null){
+        Account accountToUse = callingAccount;
+        if (accountName != null && domainId != null) {
+            try {
+                accountToUse = _accountService.getAccount(_accountService.finalyzeAccountId(accountName, domainId, projectId, true));
+
+            } catch (Exception exception) {
```

Comment:
Do we need a catch all?

Reply:
You are right, should have included specific exception. I have updated PR, please review it.